js/src/components/paid-ads/budget-section/index.js

```diff
@@ -13,30 +13,57 @@ import BudgetRecommendation from './budget-recommendation';
 import useGoogleAdsAccount from '.~/hooks/useGoogleAdsAccount';
 import AppInputPriceControl from '.~/components/app-input-price-control';
 
+/**
+ * Renders <Section> and <Section.Card> UI with campaign budget inputs.
+ *
+ * @param {Object} props React props.
+ * @param {Object} props.formProps Form props forwarded from `Form` component.
+ * @param {boolean} [props.disabled=false] Whether display the Card in disabled style.
+ */
 const BudgetSection = ( props ) => {
 	const {
 		formProps: { getInputProps, values },
+		disabled = false,
 	} = props;
 	const {
 		country: [ selectedCountryCode ],
 		amount,
 	} = values;
 	const { googleAdsAccount } = useGoogleAdsAccount();
-	const monthlyMaxEstimated = getMonthlyMaxEstimated( values.amount );
+
+	const amountValue = disabled ? '' : values.amount;
+	const monthlyMaxEstimated = disabled
+		? ''
+		: getMonthlyMaxEstimated( values.amount );
```

Comment:
❓  So here the form value `amount` doesn't change, and we conditionally display things based on `disabled`. 

I'm not sure if this might be a problem or not. When users clear the list of countries and then type a new country, the previous daily average cost value reappear. See a demo video below:

https://user-images.githubusercontent.com/417342/155393375-4e8390a0-e9c7-4204-824b-85fcf2a00621.mov

I'm thinking it might be better if we update the form value `amount` to e.g. `undefined` when the countries are removed? If we do that, then I think we don't need to have these 4 lines of code. What do you think? 🤔

Reply:
I think it depends on which scenario is the priority from the UI design/user perspective.

- As a user, I expect the form content I filled to be cleared only when I actively reset it.
- As a user, I expect when I fill a form, if an important field changes, the related important fields must be reset, so that I can be prompted to rethink and confirm that those field values are correct by filling them again.

But I can't tell which one or any others is exactly the priority. 😅 

---

Hey @j111q 👋 , could you help with inputting which is the preferred one or other options?

Reply:
Hmmm yeah thanks for the question and agree with the two competing priorities you stated Eason! 🤔 

I think for this one, it's cleaner to prioritise (2): `As a user, I expect when I fill a form, if an important field changes, the related important fields must be reset, so that I can be prompted to rethink and confirm that those field values are correct by filling them again.`

I don't anticipate many users removing all their countries after they've already filled in the amount -- but if that does happen, let's reset the form for them, and make it clear that they need to fill in the budget again just in case it has changed. 👍

Reply:
> I think for this one, it's cleaner to prioritise (2): `As a user, I expect when I fill a form, if an important field changes, the related important fields must be reset, so that I can be prompted to rethink and confirm that those field values are correct by filling them again.`


@j111q thanks for replying!

@ecgan, I updated the behavior to this scenario in https://github.com/woocommerce/google-listings-and-ads/pull/1268/commits/e07c7edac700e0161dfbbc806839841771d7bca7. Could you help with second round review?

![Kapture 2022-02-24 at 15 26 24](https://user-images.githubusercontent.com/17420811/155477956-e4dbcd8a-4a51-4fe4-8f6d-0dd6ad6a442b.gif)

Reply:
@eason9487 , I reviewed the code and left my comment in https://github.com/woocommerce/google-listings-and-ads/pull/1268#discussion_r814090894.